mlir/include/mlir/Dialect/Linalg/IR/LinalgInterfaces.h

```diff
@@ -30,6 +30,17 @@ class IteratorTypeAttr;
 class LinalgOp;
 class GenericOp;
 
+/// Container for result values of decomposition.
+/// - `decomposedOps` contains operations created by the decomposition that are
+/// returned to the caller for further transformations.
+/// - `decomposedValues` contains the values corresponding to the result of the
+/// aggregate operation.
+struct DecompositionResult {
+  /// TODO: can be further constrained to LinalgOp.
+  SmallVector<Operation *> decomposedOps;
+  SmallVector<Value> decomposedValues;
```

Comment:
Are these effectively replacements for the original results of the operation. `decomposedValues` is a misleading name. Maybe call it `replacements`?

Reply:
done